steam-wrapper

```diff
@@ -18,4 +19,38 @@ if [ -d $_STEAM_HOME ]; then
     set +x
 fi
 
+# Workaround for the fact many games, and Steam itself,
+# hardcode xdg paths to the defaults.
+_STEAM_BASE="$HOME/.var/app/com.valvesoftware.Steam"
+_CONFIG=$_STEAM_BASE/.config
+_DATA=$_STEAM_BASE/.local/share
+_CACHE=$_STEAM_BASE/.cache
+
+mkdir -p $_CACHE $_CONFIG $_DATA
+
+# Migrate any old data:
+if [ ! -L $XDG_CONFIG_HOME ]; then
+  cp -a $XDG_CONFIG_HOME/* $_CONFIG
+  # You can't delete $XDG_CONFIG_HOME/user-dirs.dirs
```

Comment:
The only thing I'm a bit concerned about with this approach is I have no idea how large user-dirs.dirs is and we're creating an extra copy here due to bind mount. The options basically are
1) Avoid copying bind-mounted things, just created directories where missing (need to immediately restart Flatpak because bind mount happens at startup)
2) Copy over everything. Everything should work from current run too, however, on next run we are bind mounting over directory with the real contents so we're wasting some space

Reply:
Its a text file with a few lines, its not a concern.